addons/azureSets/azuresets.lua

```diff
@@ -68,17 +70,25 @@ windower.register_event('login', initialize)
 
 windower.register_event('job change', initialize:cond(function(job) return job == 16 end))
 
-function set_spells(spellset)
+function set_spells(spellset, setmode)
     if windower.ffxi.get_player()['main_job_id'] ~= 16 --[[and windower.ffxi.get_player()['sub_job_id'] ~= 16]] then return nil end
     if settings.spellsets[spellset] == nil then return end
-    if settings.spellsets[spellset]:equals(get_current_spellset()) then
-        error(spellset..' was already equipped.')
+    if is_spellset_equipped(settings.spellsets[spellset]) then
+        log(spellset..' was already equipped.')
         return
     end
 
     log('Starting to set '..spellset..'.')
-    set_spells_from_spellset(spellset, 'remove')
-    return
+    if setmode:lower()=='clearfirst' then
+        remove_all_spells()
+        set_spells_from_spellset:schedule(settings.setspeed, spellset, 'add')
+    else -- if setmode:lower() == 'preservetraits' then
```

Comment:
I originally had that if there but then realized that misspelling either would then do nothing. Looking back, I think it should warn the user if something unexpected happens and do nothing.